sceptre/resolvers/no_value.py

```diff
@@ -0,0 +1,14 @@
+from sceptre.resolvers import Resolver
+
+
+class NoValue(Resolver):
+    """This resolver resolves to nothing, functioning just like the AWS::NoValue special value. When
+    assigned to a resolvable Stack property, it will remove the config key/value from the stack or
+    the container on the stack where it has been assigned, as if this value wasn't assigned at all.
+
+    This is mostly useful for simplifying conditional logic on Stack and StackGroup config files
+    where, if a certain condition is met, a value is passed, otherwise it's not passed at all.
+    """
+
+    def resolve(self) -> None:
+        return None
```

Comment:
Would you get the same affect when using this resolver as passing in `None`?  like this..
```
sceptre_user_data:
  MyParam: "None"
```

will the above also remove `MyParam` key? or does that only happen when you set `MyParam: !no_value`?

Reply:
No, you technically wouldn't. `None` in yaml resolves to the string `"None"` in Python. It's effectively the same, though, as `~` and `null`.

![image](https://user-images.githubusercontent.com/77296393/146842630-6c9d8e5b-18ef-49b2-acac-c7a8b1f91149.png).

Reply:
>will the above also remove MyParam key? or does that only happen when you set MyParam: !no_value?

No, `"None"` will not result in `MyParam` from being removed. 

```
sceptre_user_data:
    MyParam: "None"

# This becomes...
{
    'sceptre_user_data': {
        'MyParam': 'None'
    }
}

sceptre_user_data:
    MyParam: null

# This becomes... 
{
    'sceptre_user_data': {
        'MyParam': None
    }
}

sceptre_user_data:
    MyParam: !no_value

# This becomes...
{
    'sceptre_user_data': {}
}
```